test/blackbox-tests/test-cases/glob_files_rec.t

```diff
@@ -82,3 +82,26 @@ Check that generated directories are ignored
 
   $ dune build @x
 
+Check that we get a user error if an absolute path is passed to glob_files_rec
```

Comment:
Separate PR if possible please.

Reply:
moved to https://github.com/ocaml/dune/pull/6331